src/scheduler/Executor.cpp

```diff
@@ -53,6 +53,13 @@ void Executor::finish()
         }
     }
 
+    // Await dead threads
+    for (auto dt : deadThreads) {
```

Comment:
Just in case, should we use `auto&` here?

Reply:
Given that it's a vector of pointers I think it's ok.